### ISSUE TITLE
Add filtering by merge request source branch

### ISSUE THREAD

**Issue by mnunezdm** [jenkinsci/gitlab-plugin#881]
### Context
- **Gitlab plugin version**: latest
- **Gitlab version**: latest
- **Jenkins version**: latest
- **Job type**: Pipeline (but should apply to all)

### Problem description

Add the possibility to filter also by source branch not only by target branch when a merge request has triggered a build
- I did a MR in the same repository
- I would like to have the possibility to filter also by the source branch and not only by the target branch
- I could not do it because that functionality is not implemented

I have a workaround to this lack of functionality aborting the job if it does not match the expected branch name, but, I have to do this in the pipeline, so I get a new build log

Thank you!
Miguel

**Comment by omehegan** [comment]
@mnunezdm this was added and released in 1.5.10. See https://github.com/jenkinsci/gitlab-plugin/pull/754.

**Comment by mnunezdm** [comment]
Hi @omehegan 

I have checked the implementation, I didn't notice that this was implemented in one of the prior versions.

The only thing I see in that implementation is that there is currently not possible to filter by both source and target branches.

We have the use case where we want to have a specific pipeline for every dev branch that goes into master. We currently have a multi-project in single repo, each dev branch requires different configuration, permissions so we need to have different jobs.

Huge thank you!

Miguel
